collect_app/src/main/java/org/odk/collect/android/widgets/ExIntegerWidget.java

```diff
@@ -73,7 +73,7 @@ public ExIntegerWidget(Context context, FormEntryPrompt prompt) {
         Integer i = getIntegerAnswerValue();
 
         if (i != null) {
-            mAnswer.setText(i.toString());
+            mAnswer.setText(String.format("%d", i.toString()));
```

Comment:
I think you need to remove the call on toString() for it to be an improvement. I'm surprised that this works -- %d doesn't require the param to be an int? I think %f is for a double.